src/Infer/Unify.hs

```diff
@@ -21,14 +21,21 @@ bind a t | t == TVar a     = return M.empty
          | otherwise       = return $ M.singleton a t
 
 
+cleanTCompMain :: String -> String
+cleanTCompMain = filter (/= '.') . reverse . takeWhile (/= '.') . reverse
```

Comment:
Do we need the filter here ?

Reply:
I could be wrong, but it looks like it's filtering with the same predicate that `takeWhile` uses, and `reverse` shouldn't be changing anything, so I would say no?